### ISSUE TITLE
[Swift Codegen] Enum is missing namespace 

### ISSUE THREAD

**Issue by kimdv** [apollographql/apollo-ios#2301]
## Bug report

Generated enum

```swift
public extension CustomerGraphQL {
  public enum ConsentType: String, EnumType {
    case GENERAL
// More cases
  }
}
```

Generated query code:

``` swift 
public class ConsentQuery: GraphQLQuery {
  public var consentType: GraphQLEnum<ConsentType>

  public init(consentType: GraphQLEnum<ConsentType>) {
    self.consentType = consentType
  }
// More code
}
```

expected code

``` swift 
public class ConsentQuery: GraphQLQuery {
  public var consentType: GraphQLEnum<CustomerGraphQL.ConsentType>

  public init(consentType: GraphQLEnum<CustomerGraphQL.ConsentType>) {
    self.consentType = consentType
  }
// More code
}
```

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: [1.0.0 Alpha 6](https://github.com/apollographql/apollo-ios/releases/tag/1.0.0-alpha.6)
- Xcode version: 13.4.1
- Swift version: 5.6.1
- Package manager: 5.6.1

## Steps to reproduce

Our generation command

```swift
  struct GenerateCode: ParsableCommand {
    static var configuration = CommandConfiguration(
      commandName: "generate",
      abstract: "Generates swift code from your schema + your operations based on information set up in the `GenerateCode` command.")

    func run() throws {
      /// The root of the target for which you want to generate code.
      let targetRootURL = SourceRootURL
        .apollo.childFolderURL(folderName: "customer-ios/graphql")

      /// The URL where the generated schema files will be written to.
      let schemaModuleURL = SourceRootURL
        .apollo.childFolderURL(folderName: "customer-ios/graphql")

      // Make sure the folders exists before trying to generate code.
      try FileManager.default.apollo.createDirectoryIfNeeded(atPath: targetRootURL.path)
      try FileManager.default.apollo.createDirectoryIfNeeded(atPath: schemaModuleURL.path)

      // Create the Codegen configuration object. For all configuration parameters see: https://www.apollographql.com/docs/ios/api/ApolloCodegenLib/structs/ApolloCodegenConfiguration/
      let codegenConfiguration = ApolloCodegenConfiguration(
        schemaName: "GraphQLApi", input: ApolloCodegenConfiguration.FileInput(schemaPath: SchemaOutputURL.path, searchPaths: ["\(targetRootURL.path)/**/*.graphql"]),
        output: ApolloCodegenConfiguration.FileOutput(
          schemaTypes: ApolloCodegenConfiguration.SchemaTypesFileOutput(
            path: schemaModuleURL.path,
            moduleType: .embeddedInTarget(name: "customer-ios"))
        )
      )

      // Actually attempt to generate code.
      try ApolloCodegen.build(with: codegenConfiguration)
    }
  }
```

**Comment by calvincestari** [comment]
Thanks for the bug report @kimdv. Is this new for 1.0.0-alpha.6, do you know if it was present in previous alpha releases?

**Comment by kimdv** [comment]
It was also present in 4 and 5 😅

**Comment by AnthonyMDev** [comment]
Thanks for this! We haven't done enough testing with the `embeddedInTarget` module type obviously!

@calvincestari When we fix this, let's make sure we only add the name space when it's needed. If the enums are generated into a schema module, they shouldn't need namespacing.

**Comment by calvincestari** [comment]
@kimdv, this should be fixed with the merging of #2330. We plan to release 1.0.0-alpha.8 very soon and this will be included.
